### ISSUE TITLE
Unpacking large archives is slow

### ISSUE THREAD

**Issue by maximelianos** [ib/xarchiver#85]
When opening a 5-gb zip archive, the unpacking is very slow. Wouldn't it be more efficient if the archiver only extracted the needed file list?

**Comment by ib** [comment]
It actually does only list files when opening an archive.

**Comment by joezhouchenye** [comment]
Just use file-roller instead. Xarchiver cannot even open the archive file on my ARM board.

**Comment by joezhouchenye** [comment]
It's using Xfce, not Gnome. I use the Thunar file manager and thunar-archive-plugin for my ARM board. Maybe Xarchiver and others have better support for regular PC. The latest Xarchiver might work on the board but I don't want to waste my time.

I just used apt to install file-roller and didn't check carefully what dependencies are installed. I don't care if a lot of Gnome-related stuff is installed as long as it works well. Flatpak can install KDE and Gnome stuff on the same PC, but people still use it. I don't think installing other DE's packages is a big problem.

**Comment by ib** [comment]
@maximelianos What exactly is slow, opening the archive (displaying the contents) or extracting files from it?

**Comment by ib** [comment]
@reukiodo What kind of archive and what size is it? How much free RAM and swap do you have?

> Package: xarchiver
> Version: 1:0.5.4.20-2

This actually is pretty much outdated, although I know that it is current debian stable.

**Comment by reukiodo** [comment]
Correct, this was on the armbian build for PineBookPro. I believe the tar.xz archive was any one of the images downloaded for the PineBookPro. Manjaro, Armbian, DietPi, etc.

**Comment by reukiodo** [comment]
I can't boot Armbian anymore to re-validate unless I remove Towboot from SPI - another bug; https://github.com/Tow-Boot/Tow-Boot/issues/319

Right now I have Manjaro installed on eMMC as that's the only distro that I've gotten Towboot to boot into.

**Comment by reukiodo** [comment]
Manjaro seems to have an even older version? 0.5.4.22
